test/ret_test.exs

```diff
@@ -342,210 +237,149 @@ defmodule RetTest do
+          model_owned_file: model_owned_file,
           name: "fake scene",
-          slug: "fake-scene-slug",
           scene_owned_file: json_owned_file,
           screenshot_owned_file: image_owned_file,
-          model_owned_file: model_owned_file
+          slug: "fake-scene-slug"
         })
 
-      project |> Ecto.Changeset.change(scene_id: scene.scene_id) |> Repo.update!()
-
-      1 = count(Project, target_account)
-      1 = count(Scene, target_account)
-      3 = count(OwnedFile, target_account)
+      project
+      |> Ecto.Changeset.change(scene_id: scene.scene_id)
+      |> Repo.update!()
 
-      assert :ok = Ret.delete_account(admin_account, target_account)
+      1 = count_where(Project, created_by_account_id: target_account.account_id)
+      1 = count_where(Scene, account_id: target_account.account_id)
+      3 = count_where(OwnedFile, account_id: target_account.account_id)
 
-      assert 0 === count(Project, target_account)
-      assert 0 === count(Scene, target_account)
-      assert 0 === count(OwnedFile, target_account)
+      assert :ok === Ret.delete_account(target_account.account_id, admin_account)
+      assert 0 === count_where(Project, created_by_account_id: target_account.account_id)
+      assert 0 === count_where(Scene, account_id: target_account.account_id)
+      assert 0 === count_where(OwnedFile, account_id: target_account.account_id)
     end
 
     test "deletes parent scenes" do
       {:ok, admin_account: admin_account} = create_admin_account("admin")
       target_account = create_account("target")
       other_account = create_account("other")
-
       project = create_project(target_account)
       scene = create_scene(project, target_account)
 
-      other_project =
-        other_account
-        |> create_project()
-        |> Ecto.Changeset.change(parent_scene_id: scene.scene_id)
-        |> Repo.update!()
-
-      other_project
+      other_account
+      |> create_project()
+      |> Ecto.Changeset.change(parent_scene_id: scene.scene_id)
+      |> Repo.update!()
       |> create_scene(other_account)
       |> Ecto.Changeset.change(parent_scene_id: scene.scene_id)
       |> Repo.update!()
 
-      1 = count(Project, target_account)
-      1 = count(Scene, target_account)
-      1 = count(Project, other_account)
-      1 = count(Scene, other_account)
-
-      assert :ok = Ret.delete_account(admin_account, target_account)
+      1 = count_where(Scene, account_id: target_account.account_id)
+      1 = count_where(Scene, account_id: other_account.account_id)
 
-      assert 0 === count(Project, target_account)
-      assert 0 === count(Scene, target_account)
-      assert 1 === count(Project, other_account)
-      assert 1 === count(Scene, other_account)
+      assert :ok === Ret.delete_account(target_account.account_id, admin_account)
+      assert 0 === count_where(Scene, account_id: target_account.account_id)
+      assert 1 === count_where(Scene, account_id: other_account.account_id)
```

Comment:
It seems that we lost the `Project` checks here. Were they needed to test this functionality?

Also, do we expect the `parent_scene_id` of `other_account`'s scene point to a now non-existent scene, or will it be nilified?

Reply:
These test was testing numerous things. Many `assert` calls were redundant. I’ve broken them up so that each test has only one logical assertion.